pkg/network/protocols/events/worker_pool.go

```diff
@@ -0,0 +1,48 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build linux_bpf
+// +build linux_bpf
+
+package events
+
+import "sync"
+
+type workerPool struct {
+	size      int
+	jobs      chan func()
+	waitGroup sync.WaitGroup
+	once      sync.Once
+}
+
+func newWorkerPool(size int) *workerPool {
+	pool := &workerPool{
+		size: size,
+		jobs: make(chan func()),
```

Comment:
That's a good point, but here in particular I think it is fine to block and I don't think we need a channel buffer here since we wait for _all__ calls to `Do` to finish in any case: https://github.com/DataDog/datadog-agent/blob/0ae31014e9e8317ed11b73c5b98acc088b903e69/pkg/network/protocols/events/batch_reader.go#L93
but I agree that if we were designing a more general-purpose worker pool library it could make sense to have a buffered channel